src/hf_hydrodata/gridded.py

```diff
@@ -1628,36 +1518,37 @@ def _convert_json_to_strings(options):
     return options
 
 
-def _get_ndarray_from_api(entry, options):
-    """
-    Constructs the query parameters from the entry and options provided.
+def _get_gridded_data_from_api(options):
+    """Get gridded_data using the remote API. Return None if running locally to /hydrodata.
 
     Parameters
     ----------
-    entry : hydroframe.data_catalog.data_model_access.ModelTableRow
-        variable to be downloaded.
-    options : dictionary
+    options : dict
         datast to which the variable belongs.
-    options : array
-        empty array to be populated with the time_values.
 
     Returns
     -------
-    data : numpy array
-        the requested data.
+    numpy array of the requested data or None if running locally.
     """
+
     run_remote = not os.path.exists(HYDRODATA)
 
     if run_remote:
+        if options.get("period") and not options.get("temporal_resolution"):
+            options["period"] = options["temporal_resolution"]
         options = _convert_json_to_strings(options)
-        q_params = _construct_string_from_qparams(entry, options)
-        gridded_data_url = f"{HYDRODATA_URL}/api/gridded-data?{q_params}"
+        options_list = [
+            f"{name}={value}" for name, value in options.items() if value is not None
+        ]
+        q_params = "&".join(options_list)
 
+        gridded_data_url = f"{HYDRODATA_URL}/api/gridded-data?{q_params}"
         try:
             headers = _get_api_headers()
             response = requests.get(gridded_data_url, headers=headers, timeout=4000)
             if response.status_code in [500, 502]:
                 # Retry because of timeout error
+                print("API timeout, performing retry.")
```

Comment:
Yes. Print statement are not good, but Reed has complained about performance that was sometimes caused by doing a retry. I left this is so it was externally visible if the retry happened. The retry happens less often since we increased the timeout to 60 seconds, but at least for now I wanted to see some message if this retry happened.

Reply:
That makes sense. Mostly just wanted to make sure it was intentional. I've committed an embarrassingly large number of accidental print statements  😅

Reply:
I changed it from a print to a warning. That also prints a message without failing, but without the print statement.